services/QuillLMS/app/models/concept_feedback.rb

```diff
@@ -53,10 +56,23 @@ def create_translation_mappings
     translation_mappings.create(english_text: )
   end
 
-  def translation(locale:) = translated_texts.find_by(locale:)&.translation
+  def translation(locale:, source_api: TranslatedText::OPEN_AI_SOURCE)
+    translations(locale:, source_api:)&.first&.translation
+  end
+
+  private def translations(locale:, source_api:)
+    translated_texts.where(locale:).ordered_by_source_api(source_api)
+  end
 
-  def translate! = Gengo::RequestTranslations.run(english_texts)
-  def fetch_translations! = translated_texts.each(&:fetch_translation!)
+  def translate!(locale:, source_api:)
+    case source_api
+    when TranslatedText::GENGO_SOURCE
+      Gengo::RequestTranslations.run(english_texts, locale)
+    when TranslatedText::OPEN_AI_SOURCE
+      OpenAI::SaveTranslatedTexts.run(english_texts)
```

Comment:
I would get rid of the wrapper object and just call this directly:
`      english_texts.each{ |text| OpenAI::SaveTranslatedText.run(text) }`